tests/cupyx_tests/scipy_tests/sparse_tests/test_index.py

```diff
@@ -0,0 +1,110 @@
+import unittest
+
+import cupy
+from cupy import sparse
+from cupy import testing
+
+import numpy
+
+import pytest
+
+
+@testing.parameterize(*testing.product({
+    'format': ['csr', 'csc'],
+    'density': [0.1, 0.4, 0.9],
+    'dtype': ['float32', 'float64', 'complex64', 'complex128'],
+    'n_rows': [25, 150],
+    'n_cols': [25, 150]
+}))
+@testing.with_requires('scipy')
+@testing.gpu
+class TestIndexing(unittest.TestCase):
+
+    def _run(self, maj, min=None):
+        a = sparse.random(self.n_rows, self.n_cols,
+                          format=self.format,
+                          density=self.density)
+
+        # sparse.random doesn't support complex types
+        # so we need to cast
+        a = a.astype(self.dtype)
+
+        maj_h = maj.get() if isinstance(maj, cupy.ndarray) else maj
+        min_h = min.get() if isinstance(min, cupy.ndarray) else min
+
+        expected = a.get()
```

Comment:
How about converting it to a dense array?
```suggestion
        expected = a.toarray()
```

Reply:
I'm hoping I don't find instances where the sparse slicing/indexing yields different results than the dense indexing, but the main reason I've been preferring to compare the cupy.sparse to scipy.sparse is so that I can more easily spot where a discrepancy occurred when the outputs don't match. For example, I've found it much easier to debug when I see a discrepancy where some of the `indptr` entries are off by 1 or some of the values are swapped than trying to inspect the dense form. 

I can make this change if absolutely necessary, but this has been my reasoning behind keeping it sparse, anyways.

Reply:
Sorry @cjnolet you're absolutely right. I was probably confused by the if sparse.isspmatrix(actual) test below. (So if we're sticking to sparse matrices throughout, do we still need that?)

Reply:
I don't believe any of the slicing supported by this PR will return anything dense. The reason for the `spmatrix` conditional is because there were some fancy indexing cases that would return dense matrices (actually, it would return a `matrix` object on the Scipy side but we don't support `matrix` in cupy so I just returned an `ndarray` instead. That's why you see the `numpy.ndarray(expected)`. I'll remove them from this PR since there's no need for that in these tests.

Reply:
I spoke too soon- In the case of scalars being used for both major and minor indices, a dense `ndarray` is returned from Cupy.

Reply:
I see, thank you for very detailed explanation. I actually think it's ok to keep if we're sure it'd be used later, so let's wait for Akifumi's comments.